plugins/org.eclipse.xtend.ide.common/src/org/eclipse/xtend/ide/common/highlighting/XtendHighlightingCalculator.java

```diff
@@ -384,6 +384,7 @@ public void acceptSemanticLineBreak(int charCount, RichStringLiteral origin, boo
 		@Override
 		public void acceptTemplateLineBreak(int charCount, RichStringLiteral origin) {
 			resetCurrentOffset(origin);
+			acceptor.addPosition(currentOffset, charCount, TEMPLATE_LINE_BREAK);
```

Comment:
I was talking about 'acceptSemanticLineBreak' and 'announceSemanticLinebreak', which computes the 'controlStructureSeen' flag.

In IDEA the lexical highlighting is using IDEA's template background color for the whole template string. So we highlight the insignificant WS (paint it greyish again) rather than the significant (as we do in Eclipse). If don't paint the newline, the background color goes to the end of the editor.

It's probably better to do it as we do in eclipse in the first place but I still think this code is strange and I'd like to understand what template-, semantic- and possible line breaks are meant to be.